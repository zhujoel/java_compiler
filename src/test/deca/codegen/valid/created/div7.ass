; Test pour savoir si la pile est pleine
	TSTO #6
	BOV pile_pleine
	ADDSP #4
; start main program
; Class Declaration
; Code de la table des méthodes de Object
	LOAD #null, R0
	STORE R0, 1(GB)
	LOAD code.Object.equals, R0
	STORE R0, 2(GB)
; Main Function
; Declaration des variables
	LOAD #0x1.0p1, R2
	STORE R2, 3(GB)
; Beginning of main instructions:
; /
	LOAD #0x1.4p2, R2
	LOAD 3(GB), R3
	DIV R2, R3
; /
	LOAD #0x1.4p2, R2
	LOAD 3(GB), R4
	DIV R4, R2
	LOAD R2, R1
	WINT
	HALT
code.Object.equals:
pile_pleine:
	WSTR "Erreur : Pile pleine."
	WNL
	ERROR
dereferencement_null:
	WSTR "Erreur : dereferencement null."
	WNL
	ERROR
tas_plein:
	WSTR "Erreur : Tas plein."
	WNL
	ERROR
debordement:
	WSTR "Erreur : debordement durant une operation arithmétique"
	WNL
	ERROR
io_error:
	WSTR "Erreur : input ou output"
	WNL
	ERROR
; end main program
