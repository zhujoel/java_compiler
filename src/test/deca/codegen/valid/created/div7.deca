{
    int a = 2.0;
    5.0 / a;
    print(5.0 / a);
}
