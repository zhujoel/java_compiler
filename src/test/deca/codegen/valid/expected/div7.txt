2.50000e0
